pkg/asset/manifests/network.go

```diff
@@ -76,6 +78,22 @@ func (no *Networking) Generate(dependencies asset.Parents) error {
 		serviceNet = append(serviceNet, sn.String())
 	}
 
+	networkType := netConfig.NetworkType
```

Comment:
This code is very different from other platform based checks. Please follow https://github.com/openshift/installer/blob/e090d19dda2ba9d80dba652e168fcc8ed54f1b55/pkg/asset/manifests/infrastructure.go#L79 for consistency.


Also move this switch after the default network object is created and override it to kuryr when the conditions are met.

Reply:
@abhinavdahiya: Uhm, do you have something like this in mind?

```golang
	switch installConfig.Config.Platform.Name() {
	case openstacktypes.Name:
		canRunKuryr := installConfig.Config.Platform.OpenStack.TrunkSupport == "1" && installConfig.Config.Platform.OpenStack.OctaviaSupport == "1"

		switch no.Config.Spec.NetworkType {
		case "Kuryr":
			if !canRunKuryr {
				return errors.New("kuryr NetworkType can only be used on OpenStack platforms with Neutron trunk ports and Octavia enabled")
			}
		case defaults.DefaultNetworkType:
			if canRunKuryr {
				no.Config.Spec.NetworkType = "Kuryr"
			} else {
				no.Config.Spec.NetworkType = "OpenShiftSDN"
			}
		}
	default:
	case "Kuryr":
		return errors.New("kuryr NetworkType can only be used on OpenStack platforms")
	case defaults.DefaultNetworkType:
			no.Config.Spec.NetworkType = "OpenShiftSDN"
	}
```

Reply:
user providing an install config with kuryr and their platform not having support seems like an validation problem and shouldn't exit here.

Reply:
```
switch installConfig.Config.Platform.Name() {
case openstacktypes.Name && kuryrPossible:
no.Config.Spec.NetworkType = "Kuryr"
default:
}
```
doing that here https://github.com/openshift/installer/blob/76cdf219b56b5d84e85f7f547fd9f16ee6094723/pkg/asset/manifests/network.go#L116 should be enough to override the choice...

Reply:
@abhinavdahiya: Uhm, sure but this is a more complicated case. We want Kuryr if it's possible *and* user have not overridden NetworkType in InstallConfig.
I'll try simplifying that by dropping the validation.